### DISCUSSION TITLE
how to build nlohmann json?

### DISCUSSION THREAD

**Discussion by js001** [nlohmann/json#2386] [Q&A] [answered]
-- Could NOT find nlohmann_json: Found unsuitable version "..", but required is at least "3.8" (found /usr/include/nlohmann)
-- Packages nlohmann_json/3.8.0 not found!
i downoaded nlohmann json with 'git clone'. And how to build in linux ( ubuntu 18.04  )?

**Comment by nlohmann** [comment]
Can you please provide more information? Which program do you try to compile? When you downloaded nlohmann/json, where did you install it?

**Reply by nlohmann** [reply]
Any news on this?

**Comment by js001** [comment, marked as answer]
how to install nlohmann json from source code in linux ubuntu? download using git clone. then how to build to install?

**Reply by nlohmann** [reply]
The installation is not specific to Ubuntu. You can use `make install` to use the library using CMake.

**Reply by kyrieSun-wow** [reply]
Sorry to bother you, how can I know the dir where the generated library is located?

**Reply by kyrieSun-wow** [reply]
I am using the following step to build the json library:
1.mkdir build
2.cd build
3.cmake ..
4.make
5.sudo make install

And I can not located the generated library,could you tell me the target DIR?Thank you.

**Reply by kyrieSun-wow** [reply]
The DIR which indicates by libdir of nlohmann_json.pc has no library.

**Reply by gregmarr** [reply]
This is a header-only library.  Are you not able to find the headers, or are you looking for `.so` or `.a` files?

**Reply by kyrieSun-wow** [reply]
Yes,I am looking for .so or  .a files.

**Reply by kyrieSun-wow** [reply]
In my project , the version of cmake can not meet the minimum_required of this nlohmann/json lib,so I try to build the library manually,and then copy the libray(.so/.a) to my project,is there anything wrong？Or,what should I do to build the lib?

**Reply by gregmarr** [reply]
There is no library.  There are only headers.

**Reply by kyrieSun-wow** [reply]
I think I know how to deploy the header-only lib,thank you!
